content/2021/2021-05/2021-05-17-pulsar-weekly.md

```diff
@@ -0,0 +1,149 @@
+
+- [Broker] Migrate the schema storage metadata to `MetadataStore`.
+  
+  https://github.com/apache/pulsar/pull/10545 ([@merlimat](https://github.com/merlimat))
+
+- [Docker] Add a new docker compose that includes all the Pulsar components.
+  
+  https://github.com/apache/pulsar/pull/10409 ([@odmarkj](https://github.com/odmarkj))
+
+### Development
+
+- [PIP-85] Add the schema information to messages in the Java Client API.
+  
+  https://github.com/apache/pulsar/pull/10476 ([@eolivelli](https://github.com/eolivelli))
+
+### Notable Feature
+
+- [Python Client] Replace `Exceptions` with `PulsarExceptions`.
+  
+  https://github.com/apache/pulsar/pull/7600 ([@lbenc135](https://github.com/lbenc135))
+
+- [Enhancement] The reader supports seeking messages from the separate message ID or time.
+  
+  https://github.com/apache/pulsar/pull/10348 ([@315157973](https://github.com/315157973))
+
+- [CLI] Support setting the time-based limit on the backlog quota through CLI.
+  
+  https://github.com/apache/pulsar/pull/10401 ([@MarvinCai](https://github.com/MarvinCai))
+
+- [Pulsar-IO] Use `Message.getReaderSchema()` in Pulsar sink connectors when possible.
+  
+  https://github.com/apache/pulsar/pull/10557 ([@eolivelli](https://github.com/eolivelli))
+
+- [Broker] Expose the topic-level `averageMsgSize` to metrics.
+  
+  https://github.com/apache/pulsar/pull/10553 ([@315157973](https://github.com/315157973))
+
+- [Enhancement] Support truncating all data of a topic without disconnecting the producers and consumers.
+  
+  https://github.com/apache/pulsar/pull/10326 ([@jangwind](https://github.com/jangwind))
+
+- [Enhancement] Support creating `MetadataCache` with the custom SerDe.
+  
+  https://github.com/apache/pulsar/pull/10543 ([@merlimat](https://github.com/merlimat))
+
+- [Broker] Get durable subscription without handling `startMessageRollbackDurationSec`.
+  
+  https://github.com/apache/pulsar/pull/10520 ([@linlinnn](https://github.com/linlinnn))
+
+- [Auth] Enable Conscrypt for Jetty in the Pulsar broker and in the Pulsar proxy.
+  
+  https://github.com/apache/pulsar/pull/10541 ([@lhotari](https://github.com/lhotari))
+
+- [Broker] Dispatch messages to consumers with permits.
+  
+  https://github.com/apache/pulsar/pull/10417 ([@rdhabalia](https://github.com/rdhabalia))
+
+- [Enhancement] Support disabling the maximum queue size for the producers.
+  
+  https://github.com/apache/pulsar/pull/9650 ([@merlimat](https://github.com/merlimat))
+
+- [Auth] Support the optional authentication method name header in HTTP authentication.
+  
+  https://github.com/apache/pulsar/pull/6799 ([@KannarFr](https://github.com/KannarFr))
+
+### Notable Bug Fix
+
+- [Broker] Fix the issue that the `AdvertisedAddress` in `PusarService` and in `conf.` is inconsistent.
+  
+  https://github.com/apache/pulsar/pull/10312 ([@315157973](https://github.com/315157973))
+
+- [Client] Fix the NPE that is thrown when an ACK grouping tracker checks duplicated message IDs.
```

Comment:
```suggestion
- [Client] Fix the NPE thrown when an ACK grouping tracker checks duplicated message IDs.
```

Reply:
I think it is OK to use the subordinate clause.